packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -138,87 +145,26 @@ function VolcanoPlot(props: VolcanoPlotProps, ref: Ref<HTMLDivElement>) {
     []
   );
 
-  /**
-   * Find mins and maxes of the data and for the plot.
-   * The standard x axis is the log2 fold change. The standard
-   * y axis is -log10 raw p value.
-   */
-
-  // Find maxes and mins of the data itself
-  const dataXMin = min(data.map((d) => Number(d.log2foldChange))) ?? 0;
-  const dataXMax = max(data.map((d) => Number(d.log2foldChange))) ?? 0;
-  const dataYMin = min(data.map((d) => Number(d.pValue))) ?? 0;
-  const dataYMax = max(data.map((d) => Number(d.pValue))) ?? 0;
-
-  // Determine mins, maxes of axes in the plot.
-  // These are different than the data mins/maxes because
-  // of the log transform and the little bit of padding, or because axis ranges
-  // are supplied.
-  let xAxisMin: number;
-  let xAxisMax: number;
-  let yAxisMin: number;
-  let yAxisMax: number;
-  const AXIS_PADDING_FACTOR = 0.05; // The padding ensures we don't clip off part of the glyphs that represent
-  // the most extreme points. We could have also used d3.scale.nice but then we dont have precise control of where
-  // the extremes are, which is important for user-defined ranges and truncation bars.
+  // Set maxes and mins of the data itself from rawDataMinMaxValues prop
```

Comment:
since `rawDataMinMaxValues` is optional, why not determine xmins and maxes from the data if `rawDataMinMaxValues` isn't supplied?
One reason is that it would duplicate a lot of logic. Are there other reasons?

Reply:
That was really the only reason. I was hoping to determine the raw max/min values once. I was inclined to make them required, which broke all the `VolcanoPlot` stories. Since I wasn't convinced they should be required, I reverted and opted for optional. If you notice in lines 172ish of the `VolcanoPlotViz`, we default to all 0s, so `rawDataMinMaxValues` will always be passed in. Perhaps it's better to just update the stories and go with a required prop?

Reply:
yeah i think just make it required. A default of 0 won't give us anything interesting on the plot, so we might as well just require that value to be passed in.